File: festival-web/src/main/java/cn/bdqfork/web/route/response/XmlResponseHandler.java
package cn.bdqfork.web.route.response;

import cn.bdqfork.web.util.XmlUtils;
import io.vertx.ext.web.RoutingContext;

/**
 * @author bdq
 * @since 2020/2/20
 */
public class XmlResponseHandler extends AbstractResponseHandler {

    @Override
    protected void doHandle(RoutingContext routingContext, Object result) throws Exception {
        routingContext.response().end(XmlUtils.toXml(result));
    }

}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/response/XmlResponseHandler.kt
package cn.bdqfork.kotlin.web.route.response

import cn.bdqfork.kotlin.web.util.XmlUtils
import io.vertx.ext.web.RoutingContext

/**
 * @author bdq
 * @since 2020/2/20
 */
class XmlResponseHandler : AbstractResponseHandler() {
    @Throws(Exception::class)
    override fun doHandle(routingContext: RoutingContext, result: Any) {
        routingContext.response().end(XmlUtils.toXml(result))
    }
}